src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -345,20 +345,33 @@ void MergeLibraryManifest (string mergedManifest)
 			var nsResolver = new XmlNamespaceManager (new NameTable ());
 			nsResolver.AddNamespace ("android", androidNs.NamespaceName);
 			var xdoc = XDocument.Load (mergedManifest);
+			var package = xdoc.Root.Attribute ("package")?.Value ?? string.Empty;
+			log.LogDebugMessage ($"Resolved {package} from {mergedManifest}");
 			foreach (var top in xdoc.XPathSelectElements ("/manifest/*")) {
 				var name = top.Attribute (AndroidXmlNamespace.GetName ("name"));
 				var existing = (name != null) ?
 					doc.XPathSelectElement (string.Format ("/manifest/{0}[@android:name='{1}']", top.Name.LocalName, XmlConvert.VerifyNCName (name.Value)), nsResolver) :
 					doc.XPathSelectElement (string.Format ("/manifest/{0}", top.Name.LocalName));
 				if (existing != null)
 					// if there is existing node with the same android:name, then append contents to existing node.
-					existing.Add (top.Nodes ());
+					existing.Add (FixupNameElements (package, top.Nodes ()));
 				else
 					// otherwise, just add to the doc.
-					doc.Root.Add (top);
+					doc.Root.Add (FixupNameElements (package,new XNode [] { top }));
 			}
 		}
 
+		IEnumerable<XNode> FixupNameElements(string packageName, IEnumerable<XNode> nodes)
+		{
+			foreach (var element in nodes.Where ( x => x is XElement).Select (x => x  as XElement)) {
+				foreach (var attr in element.Attributes ()) {
```

Comment:
Should this be done for *all* attribute values? Or just a select set of values, like `android:name`?

For example, if someone (mistakenly?) has `[Activity(Permission=".something")]`, which sets [`//activity/@android:permission`](https://developer.android.com/guide/topics/manifest/activity-element.html#prmsn), should that be altered? Or should that be left as-is (probably to become an `aapt` error)?

I'm inclined to think that we should only do this on select attributes:

* `@android:name`
* `@android:parentActivityName`
* `@android:targetActivity`
* `@android:backupAgent`
* `@android:manageSpaceActivity`

Uncertain about:

* `@android:requiredAccountType`
* `@android:restrictedAccountType`
* `android:targetPackage` (in `<instrumentation/>`)
* Should `//category/@android:name` be updated? Meaning, do we need to check for *element names* as well as just attributes?
* Ditto [`//action/@android:name`](https://developer.android.com/guide/topics/manifest/action-element.html).
* Eep, and [`//meta-data/@android:name`](https://developer.android.com/guide/topics/manifest/meta-data-element.html#nm)!
    * The documentation here has an example for `android:name="zoo"`. If this is truly arbitrary, meaning user code does the lookups, "auto-fixing" `android:name=".zoo"` would result in breaking code!

Finally, i haven't gone through every attribute on every element; i gave up at `<permission-tree/>`, after seeing the above "questionable" attributes.